sealing/enclave/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+#![no_std]
+#![deny(unsafe_op_in_unsafe_fn)]
+
+use std::{eprintln, slice};
+
+use sealing::{Nonce, SecretKey};
+use sgx_keys::{SgxKeyPolicy, SgxSecretBuilder};
+use sgx_rand::{Rng, SgxRng};
+use sgx_tstd as std;
+use sgx_types::sgx_status_t;
+
+/// Seal private user data.
+#[no_mangle]
+pub unsafe extern "C" fn seal_user_data(
+    // input parameters
+    user_id: *const u8,
+    user_id_length: usize,
+    // dual-purpose parameters
+    user_data: *mut u8,
+    // output parameters
+    sealed_user_data_size: usize,
+    sealed_user_sealing_key: *mut u8,
+) -> sgx_status_t {
+    let user_id = unsafe { slice::from_raw_parts(user_id, user_id_length) };
+    let user_data_buf = unsafe { slice::from_raw_parts(user_data, sealed_user_data_size) };
+
+    let mut rng = SgxRng::new().expect("SGX: RDRAND instruction failed");
+    let mut user_sealing_key = SecretKey::new([0u8; 32]);
+    rng.fill_bytes(user_sealing_key.as_mut());
+
+    let user_data_nonce = Nonce::new([0u8; Nonce::SIZE]);
+    let sealed_user_data =
+        match sealing::seal(user_data_buf, user_sealing_key, user_data_nonce, user_id) {
+            Ok(sealed_data) => sealed_data,
+            Err(_) => {
```

Comment:
I don't expect the seal function would reveal sensitive info, easpecially since the error ultimately comes from `ring_compat::aead::Error`, which I believe practices good hygiene :)